src/renderers/WebGLRenderer.js

```diff
@@ -190,11 +191,15 @@ function WebGLRenderer( parameters ) {
 		_canvas.addEventListener( 'webglcontextlost', onContextLost, false );
 		_canvas.addEventListener( 'webglcontextrestored', onContextRestore, false );
 
-		_gl = _context || _canvas.getContext( 'webgl', contextAttributes ) || _canvas.getContext( 'experimental-webgl', contextAttributes );
+		var webglVersion = _webgl2 && typeof WebGL2RenderingContext !== 'undefined' ? 'webgl2' : 'webgl';
```

Comment:
Do you think we should try to create WebGLContext if we fail to create WebGL2Context?

Reply:
I think we can avoid a fallback if the use explicitly requests WebGL2 with a parameter. Compatibility can be verified on app level. Maybe we can enhance [Detector](https://github.com/mrdoob/three.js/blob/dev/examples/js/Detector.js) in order to support the user with this task. Something like: 
```js
if ( Detector.webgl2 ) // use WebGL 2
```

Reply:
+1 for `Detector.webgl2`

Reply:
Added. Thanks!

Reply:
I agree with @Mugen87 I think that if we explicitly request WebGL2 context and it's not available, it shouldn't fallback to webgl1. Probably someone requesting `webgl2` would like to use specific features that will fail on webgl1.
We could decide in the future if use by default webgl2 if available and we don't specify any value.

Reply:
I added two options. `webgl2`: force to create WebGL2Context, `autoWebgl2`: create WebGL2Context if browser seems to support WebGL 2.0 otherwise create WebGLContext.

We may want autoWebgl2 as the default once Three.js + WebGL 2.0 becomes stable (See https://github.com/mrdoob/three.js/pull/13701#issuecomment-377009238)

Reply:
I would just leave one option. Something like:
* webgl2: false -> webgl1
* webgl2: true -> webgl2 or throw error if not possible
* webgl2: 'auto' ? -> try webgl2 or fallback to webgl1.
* webgl2: undefined -> right now it should be use webgl1, once webgl2 will be stable we could do the same as `auto`.

Reply:
Combining the two options into a single one is nice. But I don't really like mix type, boolean or strings.

So how about `webglVersion` option,
- 'webgl': webgl1
- 'webgl2': webgl2
- 'auto': webgl2 if browser seems to support WebGL 2.0, otherwise webgl1
- undefined: right now acts as 'webgl', once Three.js + WebGL 2.0 becomes stable acts as 'auto'

Reply:
@takahirox yeah! I like that approach better :+1:

Reply:
Updated.